tensorflow/python/ops/clip_ops.py

```diff
@@ -56,18 +57,10 @@ def clip_by_value(t, clip_value_min, clip_value_max,
   """
   with ops.name_scope(name, "clip_by_value",
                       [t, clip_value_min, clip_value_max]) as name:
-    t = ops.convert_to_tensor(t, name="t")
-
-    # Go through list of tensors, for each value in each tensor clip
-    t_min = math_ops.minimum(t, clip_value_max)
-    # Assert that the shape is compatible with the initial shape,
-    # to prevent unintentional broadcasting.
-    _ = t.shape.merge_with(t_min.shape)
-
-    t_max = math_ops.maximum(t_min, clip_value_min, name=name)
-    _ = t.shape.merge_with(t_max.shape)
-
-  return t_max
+    return gen_math_ops._clip_by_value(t,
+                                       clip_value_min,
+                                       clip_value_max,
+                                       name=name)
```

Comment:
Before your change, TF was automatically generating a gradient from the gradient of the maximum and minimum ops used to clip the value, but it can't do this for your new ClipByValue: therefore you need to define and register a gradient op for ClipByValue.

Reply:
Thanks. The gradient registration for ClipByValue has been added.